### DISCUSSION TITLE
`Image<Rgba32>.GetPixelSpan()` from 1.0.0-beta0007 to 2.1.2

### DISCUSSION THREAD

**Discussion by avengerx** [SixLabors/ImageSharp#2130] [Ideas] [answered]
In 1.0.0-beta0007 I could get an array of an image with:

```csharp
Image<Rgba32> img = SixLabors.ImageSharp.Image.Load<Rgba32>(ImagePath);
Rgba32[] Colors = img.GetPixelSpan().ToArray();
```

Nice!

But then, trying to upgrade to 2.1.2, I see this method been reworked once to `TryGetPixelSpan(out Span<Rgba32>)` (#1263), and nowadays it is also gone.

Digging a bit about in Stackoverflow I found this answer: https://stackoverflow.com/questions/50025908/convert-imagergba32-to-byte-using-imagesharp#answer-63758770

That was the closest I found to what should be the right way to go... But then, there's quite a fight in this answer's comments section. I felt it was the right thing to post this here to see if there's a best practice now to do the same in the current, 2.1.2, version of the library. Any idea?

**Comment by avengerx** [comment, marked as answer]
For now it seems re-implementing it works via this extension method in my project:

```csharp
public static Span<T> GetPixelSpan<T>(this Image<T> imageRef) where T : unmanaged, IPixel<T>
{
    var memoryFootprint = new T[imageRef.Width * imageRef.Height];
    var pixelSpan = new Span<T>(memoryFootprint);
    imageRef.CopyPixelDataTo(pixelSpan);
    return pixelSpan;
}
```

Looking forward to everyone's insight on this matter. And if that's adequate an approach, I hope this helps everyone else facing this issue.

**Reply by antonfirsov** [reply]
Since ImageSharp 2.0+ any significantly large image is discontigous in memory, so `image.GetPixelSpan()` would fail to grab a single span most of the time, therefore we decided to remove that method. Instead we are exposing `CopyPixelDataTo()`, so you took the correct approach.

**Reply by avengerx** [reply]
Thanks for the quick response and feedback, glad the approach was adequate on the first few shots! I could really not find anything helpful about this around or on google.

You guys removed that method just because it could throw exception if runtime couldn't allocate the memory?

I mean... wouldn't the developer be able to handle it in that case given their project constraints? It sounds unfair, but I'm really curious on the reasons you guys had to to go out of your way and wipe the implementation off the code. :)
